src/components/Heading/Heading.tsx

```diff
@@ -24,24 +31,33 @@ export interface HeadingProps {
   urlClass?: string;
 }
 
-export default function Heading(props: React.PropsWithChildren<HeadingProps>) {
+// Used to map between ButtonTypes enum values and Chakra variant options.
+const variantMap = {};
+for (const type in HeadingDisplaySizes) {
+  variantMap[HeadingDisplaySizes[type]] = HeadingDisplaySizes[type];
+}
+
+/**
+ * Map the HeadingDisplaySizes to the Heading Chakra theme variant object. If a wrong
+ * value is passed (typically in non-Typescript scenarios), then the default
+ * is the "primary" variant.
+ */
+const getVariant = (displaySize) => variantMap[displaySize] || null;
+
+function Heading(props: React.PropsWithChildren<HeadingProps>) {
   const {
-    blockName,
     className,
     displaySize,
     id,
-    level,
-    modifiers = [],
+    level = HeadingLevels.Two,
     text,
     url,
     urlClass,
   } = props;
 
-  const baseClass = "heading";
-
-  if (displaySize) {
-    modifiers.push(displaySize);
-  }
+  const variant = displaySize ? getVariant(displaySize) : `h${level}`;
+  const styles = useStyleConfig("Heading", { variant });
+  const asHeading: any = `h${level}`;
```

Comment:
You can type `asHeading` as `As`: `const asHeading: As = ....` and import the `As` type from chakra.

Reply:
That's for pointing this out.